### DISCUSSION TITLE
Unable to connect metamask with Zksync

### DISCUSSION THREAD

**Discussion by RoccoAlfano** [Cyfrin/foundry-full-course-cu#2172] [General]
Every time I push the connect button on chainlist for zksync nothing will happen, prior to that I had a coinbase wallet in my browser which it would pull up so I removed it from the browser but still it did not work.

**Comment by EngrPips** [comment]
What wallet are you using in your browser?

**Comment by Hajali-Selim** [comment]
Hello, I'm having the same issue, Chainlist doesn't seem to interact with Metamask. I tried connecting Metamask to the testnet manually using the info found [here](https://rpc.info/zksync-sepolia) but I always get the following error.

![metamask_error](https://github.com/user-attachments/assets/781ec78c-4b3f-4969-96a3-0c274161a7d2)

Chainlist does recognize my Exodus wallet (like Coinbase, in your case) when I click 'Connect Wallet' but it is irrelevant here, since - I guess - neither support zkSync Sepolia testnet.

I also tried bridging 0.01 Sepolia ETH into zkSync Sepolia testnet anyway, and they got lost.

**Reply by EngrPips** [reply]
I use a Metamask wallet and adding zkSync chain to my Metamask was a breeze. Maybe try Metamask wallet.

**Reply by Hajali-Selim** [reply]
I am using a Metamask wallet, that's how that window opened up (attached image), unless I'm missing something.
It has the option of connecting to zkSync Era Mainnet but I can't figure out how to connect to the zkSync Sepolia Testnet, which is the exercise in that course.

**Reply by EngrPips** [reply]
Hello, I re-add the `zkSync Sepolia` chain to my metamask again, and I didn't face any issue whatsoever. Below are the images that show my step by step process

<img width="1440" alt="Screenshot 2024-08-07 at 10 10 02" src="https://github.com/user-attachments/assets/2c882247-64b8-4047-b5ac-16802e1bf666">

<img width="1440" alt="Screenshot 2024-08-07 at 10 10 59" src="https://github.com/user-attachments/assets/5572a6eb-1075-4e1c-8b47-26b98dfe3017">

<img width="1440" alt="Screenshot 2024-08-07 at 10 10 12" src="https://github.com/user-attachments/assets/c1b87be3-bb46-4d27-8e41-f44dadcefd80">

<img width="1440" alt="Screenshot 2024-08-07 at 10 11 07" src="https://github.com/user-attachments/assets/fdc85063-06f6-4216-9edc-20598bfa5bb2">

**Reply by Hajali-Selim** [reply]
Thanks a lot for your help, but I don't get that 3rd screen that suggests to "allow the site to switch the network" from Sepolia to zkSync Sepolia Testnet. Instead I get the error I shared above (network not recognized by Metamask).
I wondered whether this could be due to my browser (Brave). So I tried on Firefox, installed Metamask there and I got the exact same error.

**Reply by Hajali-Selim** [reply]
Update! It works now, I have no idea what changed but now, the network did switch to Testnet 😄

**Reply by EngrPips** [reply]
I am glad you figured this out. Keep crushing it, buddy!

**Reply by RoccoAlfano** [reply]
Been away for a couple days so was unable to reply, I fixed the issue also cheers!!

**Reply by EngrPips** [reply]
Awesome, I am glad you fixed the issue. Keep crushing it.

**Reply by Blazingkevin** [reply]
I am facing this same issue, I click on the button to "Connect Wallet" and it does nothing

**Reply by EngrPips** [reply]
> I am facing this same issue, I click on the button to "Connect Wallet" and it does nothing

Try other `rpc-url` on the Chainlist website.
